api/src/test/java/ai/djl/modality/audio/AudioFactoryTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.modality.audio;
+
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.NDManager;
+import ai.djl.ndarray.types.Shape;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class AudioFactoryTest {
+
+    private static final String PATH = "src/test/resources/speech.wav";
+    private static final float[] DATA = {0.001f, 0.002f, 0.003f};
+
+    private AudioFactory factory;
+
+    @BeforeTest
+    public void setUp() {
+        factory = AudioFactory.getInstance();
+    }
+
+    @Test
+    public void testFromFile() throws IOException {
+        Audio audio = factory.fromFile(Paths.get(PATH));
+        Assert.assertEquals(audio.getSampleRate(), 8000f);
+        Assert.assertEquals(audio.getChannels(), 1);
+    }
+
+    @Test
+    public void testFromUrl() throws IOException {
+        Audio audio = factory.fromUrl(PATH);
+        Assert.assertEquals(audio.getSampleRate(), 8000f);
+        Assert.assertEquals(audio.getChannels(), 1);
+    }
+
+    @Test
+    public void testFromInputStream() throws IOException {
+        try (InputStream is = new BufferedInputStream(Files.newInputStream(Paths.get(PATH)))) {
```

Comment:
Does BufferedInputStream required? If true, we need add this in our API

Reply:
Yes BufferedInputStream is needed. I have added it to API.